server/controllers/finance/reports/cashReport/index.js

```diff
@@ -0,0 +1,294 @@
+/**
+ * cashReport Controller
+ *
+ *
+ * This controller is responsible for processing cash report.
+ *
+ * @module finance/cashReport
+ *
+ * @requires lodash
+ * @requires lib/db
+ * @requires lib/ReportManager
+ * @requires lib/errors/BadRequest
+ */
+
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+const BadRequest = require('../../../../lib/errors/BadRequest');
+
+const TEMPLATE_1 = './server/controllers/finance/reports/cashReport/report_combined.handlebars';
+const TEMPLATE_2 = './server/controllers/finance/reports/cashReport/report_separated.handlebars';
+
+// expose to the API
+exports.document = document;
+
+function getOpeningBalance(accountId, maxDate) {
+  const sql = `
```

Comment:
For performance reasons, this would be much more efficient to use `period_total` to pick up the balances from period periods.  In fact, this function already exists in [server/controllers/finance/accounts/extra.js](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/accounts/extra.js#L124).  It has the signature `function getOpeningBalanceForDate(accountId, date)`.  It would be best to use that function since it will use `period_total` over `general_ledger` where possible.